lib/ClassUtils.lua

```diff
@@ -26,28 +33,32 @@ function ClassUtils.makeClass(name, constructor)
 	return Class
 end
 
-function ClassUtils.makeConstructedClass(name, constructor)
-	constructor = constructor or function()
-		end
-	local Class
-	Class =
+function ClassUtils.makeClassWithInterface(name, interface)
+	local function getImplementsInterface(currentInterface)
+		local ok, problem = tea.values(tea.callback)(currentInterface)
+		assert(ok, string.format([[Class %s does not have a valid interface
+%s]], name, problem or ""))
```

Comment:
I normally do `tostring(problem)` in case it is a number or something else which makes `%s` barf